suitcase/tiff/__init__.py

```diff
@@ -1,25 +1,256 @@
+    """
+    def __init__(self, directory, file_prefix='{uid}-', **kwargs):
+
+        if isinstance(directory, (str, Path)):
+            self.manager = suitcase.utils.MultiFileManager(directory)
+        else:
+            self.manager = directory
+
+        self.artifacts = self.manager._artifacts
+        self._stream_names = {}  # maps stream_names to each descriptor uids
+        self._files = defaultdict(dict)  # maps stream_name to field/file dicts
+        self._filenames = {}  # map stream_name to file names of tiff files
+        self._file_prefix = file_prefix
+        self._templated_file_prefix = ''
+        self._kwargs = kwargs
+        self._start_found = False
+
+    def start(self, doc):
+        '''Extracts `start` document information for formatting file_prefix.
+
+        This method checks that only one `start` document is seen and formats
+        `file_prefix` based on the contents of the `start` document.
+
+        Parameters:
+        -----------
+        doc : dict
+            RunStart document
+        '''
+
+        # raise an error if this is the second `start` document seen.
+        if self._start_found:
+            raise RuntimeError(
+                "The serializer in suitcase.tiff expects documents from one "
+                "run only. Two `start` documents where sent to it")
+        else:
+            self._start_found = True
+
+        # format self._file_prefix
+        self._templated_file_prefix = self._file_prefix.format(**doc)
+
+        # return the start document
+        return doc
+
+    def descriptor(self, doc):
+        '''Use `descriptor` doc to map stream_names to descriptor uid's.
+
+        This method usess the descriptor document information to map the
+        stream_names to descriptor uid's.
+
+        Parameters:
+        -----------
+        doc : dict
+            EventDescriptor document
+        '''
+        # extract some useful info from the doc
+        stream_name = doc.get('name')
+        self._stream_names[doc['uid']] = stream_name
+
+        # return the descriptor doc
+        return doc
+
+    def event_page(self, doc):
+        '''Add event page document information to a ".tiff" file.
+
+        This method adds event_page document information to a ".tiff" file,
+        creating it if nesecary.
+
+        .. warning::
+
+            All non 2D 'image like' data is explicitly ignored.
+
+        .. note::
+
+            The data in Events might be structured as an Event, an EventPage,
+            or a "bulk event" (deprecated). The DocumentRouter base class takes
+            care of first transforming the other repsentations into an
+            EventPage and then routing them through here, so no further action
+            is required in this class. We can assume we will always receive an
+            EventPage.
+
+        Parameters:
+        -----------
+        doc : dict
+            EventPage document
+        '''
+        event_model.verify_filled(doc)
```

Comment:
I guess we don't need the whole document to be filled, only the columns that we're about to try to export. Maybe we should extend this function to take an `include=[fields]` argument?

Reply:
This is true, but pending an update to event_model (let me know if you would like me to add this) I think this is the best option.

Reply:
I think we should do that soon-ish but it doesn't have to hold up with PR. Can you open an issue on `event-model` to add this feature and then an `suitcase-tiff` to refactor this once that feature is merged?